backend/danswer/connectors/linear/connector.py

```diff
@@ -0,0 +1,209 @@
+import os
+from datetime import datetime
+from datetime import timezone
+from typing import Any
+
+import requests
+
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput
+from danswer.connectors.interfaces import LoadConnector
+from danswer.connectors.interfaces import PollConnector
+from danswer.connectors.interfaces import SecondsSinceUnixEpoch
+from danswer.connectors.models import Document
+from danswer.connectors.models import Section
+from danswer.utils.logger import setup_logger
+
+logger = setup_logger()
+
+_NUM_RETRIES = 5
+
+
+def _make_query(request_body: dict[str, Any], api_key: str) -> requests.Response:
+    headers = {
+        "Authorization": api_key,
+        "Content-Type": "application/json",
+    }
+
+    response: requests.Response | None = None
+    for i in range(_NUM_RETRIES):
+        try:
+            response = requests.post(
+                "https://api.linear.app/graphql",
+                headers=headers,
+                json=request_body,
+                timeout=60,
+            )
+            if not response.ok:
+                raise RuntimeError(
+                    f"Error fetching issues from Linear: {response.text}"
+                )
+
+            return response
+        except Exception as e:
+            if i == _NUM_RETRIES - 1:
+                raise e
+
+            logger.warning(f"A Linear GraphQL error occurred: {e}. Retrying...")
+
+    raise RuntimeError(
+        "Unexpected execution when querying Linear. This should never happen."
+    )
+
+
+class LinearConnector(LoadConnector, PollConnector):
+    def __init__(
+        self,
+        batch_size: int = INDEX_BATCH_SIZE,
+    ) -> None:
+        self.batch_size = batch_size
+
+    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
+        self.linear_api_key = credentials["linear_api_key"]
```

Comment:
Should this be initialized to None in init?

Reply:
Done!